src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/IncludeMatchSeedRewriter.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Health.Fhir.Core.Features.Search.Expressions;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors
+{
+    /// <summary>
+    /// Rewriter used to add an All SearchParamTableExpression to serve as a seed for match results when the SearchParamTableExpressions in a SqlRootExpression
```

Comment:
>seed for match [](start = 77, length = 14)

I see that 'seed' in few other places, but i'm not sure I quite understand idea behind it. Any chance you can comment what exactly seed is?

Reply:
Yeah, we should improve the comments here. 

Suppose you have an include query like `Observation?code=abc&_include=Observation:subject`. The include will be based on the CTE that gets the IDs of the first 10 observations where the code is `abc`.

Now suppose you have simply `Observation?_include=Observation:subject`. Here there is no CTE to base the include on, so we inject one that yields all Observations (well, the first 10). This is what I'm calling "seeding" the include.